coalib/bearlib/languages/definitions/Haskell.py

```diff
@@ -0,0 +1,20 @@
+from coalib.bearlib.languages.Language import Language
+
+
+@Language
+class Haskell:
+    extensions = '.hs', '.lhs',
+    versions = 1.0, 1.1, 1.2, 1.3, 1.4, 98, 2010
+    special_chars = list('!#$%&*+./<=>?@\\^|-~')
+    comment_delimiters = '--',
+    multiline_comment_delimiters = {'{-': '-}'}
+    keywords = [
+        '!', '-', '--', '-<', '-<<', '->', '::', ';',
+        '<-', ',', '=', '=>', '>', '?', '#', '*', '@', '[|, |]',
+        '_', '`', '{, }', '{-, -}', '|', '~', 'case', 'of',
```

Comment:
many of these belong in `encapsulators`, which you have omitted from this language definition.

And as mentioned by @shashwatmishra76 , we do not need to repeat each `special_chars` here as keywords.